src/lib/bootloader/proposal_client.rb

```diff
@@ -226,7 +224,23 @@ def single_click_action(option, value)
 
       value ? stage1.add_udev_device(device) : stage1.remove_device(device)
 
+      update_required_packages
+
       Yast::Bootloader.proposed_cfg_changed = true
     end
+
+    def update_required_packages
+      bl = ::Bootloader::BootloaderFactory.current
+      bootloader_resolvables = Yast::PackagesProposal.GetResolvables("yast2-bootloader", :package)
+
+      if (bl.name == "none" || bl.stage1.devices.empty?) &&
+          !bootloader_resolvables.empty?
+        log.info "packages #{bl.packages} are no longer required"
+        Yast::PackagesProposal.RemoveResolvables("yast2-bootloader", :package, bl.packages)
+      elsif bootloader_resolvables.empty? && bl.name != "none" && !bl.stage1.devices.empty?
+        log.info "propose to install #{bl.packages}"
+        Yast::PackagesProposal.AddResolvables("yast2-bootloader", :package, bl.packages)
+      end
```

Comment:
in general why does not use simple

```
Yast::PackagesProposal.RemoveResolvables("yast2-bootloader", :package, bootloader_resolvables)
Yast::PackagesProposal.AddResolvables("yast2-bootloader", :package, bl.packages)
```

for all scenarios? this way it will also fix switching between variants with different packages and it is clear code. Deselect old ones and select new packages.